Update starter and starter-vite-tsc examples

alenkis opened the pull request:
Updates `starter-vite-tsc` as mentioned in https://github.com/Effect-TS/examples/issues/2

- Updates `@effect` deps to latest
- adds `prettier` config for consistency
- adds `lib.ts` for convenience

I can update the other one's if these changes make sense

alenkis commented on starter-vite-tsc/src/App.tsx:
```diff
@@ -1,11 +1,17 @@
-import { Effect } from "effect/io";
+import { Effect } from "~/lib";
+
+const program = Effect.sync(() => {
+  console.log("Hello, World!");
+});
```
I thought this was a simpler example than tracing

alenkis commented on starter-vite-tsc/.prettierrc.cjs:
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  printWidth: 80,
+  singleQuote: false,
+  trailingComma: "all",
+  semi: true,
+  arrowParens: "always",
+}
```
just something that seemed reasonable, and caused the smallest diff. 
Open to changing these settings

gcanti commented:
Thank you @alenkis, we have recently released `effect` v2.0.0-next.16 (Announcement: [link](https://discord.com/channels/795981131316985866/1125094089281511474/1129269312281124944)). Since this example (along with `starter`) is linked from the [docs](https://www.effect.website/examples/starter-vite-tsc), it's now important that all imports are from the `effect` package rather than individual modules.

alenkis commented:
thanks @gcanti!

I've updated `starter-vite-tsc` and `starter` to the newest release, removed all the individual effect modules and updated other deps too. 
I've also realized that the `~/lib` pattern no longer makes sense so I've removed it. It's much more convenient to just directly import from `effect` now with the newest release 👍

gcanti commented:
> I've also realized that the ~/lib pattern no longer makes sense so I've removed it

:+1:

p.s.

Using `pipe` is fine, just FYI there's also a `.pipe()` method on all data types now:

```ts
program.pipe(
  Effect.provideLayer(NameServiceLive),
  Effect.tapErrorCause(Effect.logCause({ level: "Error" })),
  Effect.runFork
);
```
